File: solutions/2179-most-beautiful-item-for-each-query/most-beautiful-item-for-each-query.cs
public class Solution {
    public int[] MaximumBeauty(int[][] items, int[] queries) {
        Array.Sort(items, (a, b) => a[0].CompareTo(b[0]));
        var queriesWithIndex = queries.Select((q, i) => new { Query = q, Index = i })
                                    .OrderBy(x => x.Query)
                                    .ToArray();
        
        int[] res = new int[queries.Length];
        int maxBea = 0;
        int j = 0;
        foreach (var query in queriesWithIndex) {
            while (j < items.Length && items[j][0] <= query.Query) {
                maxBea = Math.Max(maxBea, items[j][1]);
                j++;
            }
            res[query.Index] = maxBea;
        }
        return res;
    }
}

File: solutions/2179-most-beautiful-item-for-each-query/most-beautiful-item-for-each-query.swift
class Solution {
    func maximumBeauty(_ items: [[Int]], _ queries: [Int]) -> [Int] {
        let sortedItems = items.sorted { $0[0] < $1[0] }
        let queriesWithIndex = queries.enumerated().map { ($1, $0) }.sorted { $0.0 < $1.0 }
        
        var res = Array(repeating: 0, count: queries.count)
        var maxBea = 0
        var j = 0
        
        for (query, index) in queriesWithIndex {
            while j < sortedItems.count && sortedItems[j][0] <= query {
                maxBea = max(maxBea, sortedItems[j][1])
                j += 1
            }
            res[index] = maxBea
        }
        return res
    }
}